internal/handlers/common/projection.go

```diff
@@ -168,286 +152,10 @@ func projectDocument(inclusion bool, doc *types.Document, projection *types.Docu
 func applyComplexProjection(k1 string, doc, projectionVal *types.Document) (err error) {
 	for _, projectionType := range projectionVal.Keys() {
 		switch projectionType {
-		case "$elemMatch":
-			var docValueA any
-			docValueA, err = doc.GetByPath(types.NewPath([]string{k1}))
-			if err != nil {
-				continue
-			}
-
-			// $elemMatch works only for arrays, it must be an array
-			docValueArray, ok := docValueA.(*types.Array)
-			if !ok {
-				doc.Remove(k1)
-				return
-			}
-
-			// get the elemMatch conditions
-			conditions := must.NotFail(projectionVal.Get(projectionType)).(*types.Document)
-
-			var found int
-			found, err = filterFieldArrayElemMatch(k1, doc, conditions, docValueArray)
-			if found < 0 {
-				doc.Remove(k1)
-			}
-		case "$slice":
-			var docValue any
-			docValue, err = doc.Get(k1)
-			if err != nil { // the field can't be obtained, so there is nothing to do
-				return
-			}
-			// $slice works only for arrays, so docValue must be an array
-			arr, ok := docValue.(*types.Array)
-			if !ok {
-				return
-			}
-			projectionVal := must.NotFail(projectionVal.Get(projectionType))
-			res, err := filterFieldArraySlice(arr, projectionVal)
-			if err != nil {
-				return err
-			}
-
-			if res == nil {
-				doc.Set(k1, types.Null)
-				return nil
-			}
-
-			doc.Set(k1, res)
 		default:
-			return NewCommandError(ErrCommandNotFound,
-				lazyerrors.Errorf("applyComplexProjection: unknown projection operator: %q", projectionType),
-			)
+			return lazyerrors.Errorf("projection of %s is not supported", projectionType)
```

Comment:
Do we need the whole switch at all if we don't put any logic?

Or alternatively, something like this (so we keep the old error as is and use "not implemented" error for the parts that are not implemented:
```go
switch projectionType {
		case "$elemMatch", "$slice":
                      return NewCommandError(ErrNotImplemented, ...))
                default:
                      return NewCommandError(ErrCommandNotFound, ...))
}
```

Reply:
Nice suggestion thanks :+1: